sdk/python/kfp/v2/compiler/compiler_test.py

```diff
@@ -70,6 +71,149 @@ def simple_pipeline(pipeline_input='Hello KFP!',):
     finally:
       shutil.rmtree(tmpdir)
 
+  def test_compile_pipeline_with_dsl_condition_should_raise_error(self):
+
+    flip_coin_op = components.load_component_from_text("""
+      name: flip coin
+      inputs:
+      - {name: name, type: String}
+      outputs:
+      - {name: result, type: String}
+      implementation:
+        container:
+          image: gcr.io/my-project/my-image:tag
+          args:
+          - {inputValue: name}
+          - {outputPath: result}
+      """)
+
+    print_op = components.load_component_from_text("""
+      name: print
+      inputs:
+      - {name: name, type: String}
+      - {name: msg, type: String}
+      implementation:
+        container:
+          image: gcr.io/my-project/my-image:tag
+          args:
+          - {inputValue: name}
+          - {inputValue: msg}
+      """)
+
+    @dsl.pipeline()
+    def flipcoin():
+      flip = flip_coin_op('flip')
+
+      with dsl.Condition(flip.outputs['result'] == 'heads'):
+        flip2 = flip_coin_op('flip-again')
+
+        with dsl.Condition(flip2.outputs['result'] == 'tails'):
+          print_op('print1', flip2.outputs['result'])
+
+      with dsl.Condition(flip.outputs['result'] == 'tails'):
+        print_op('print2', flip2.outputs['results'])
+
+    with self.assertRaises(NotImplementedError) as cm:
+      compiler.Compiler().compile(flipcoin, 'output.json')
+
+    self.assertEqual('dsl.Condition is not yet supported in KFP v2 compiler.',
+                     str(cm.exception))
+
+  def test_compile_pipeline_with_dsl_exithandler_should_raise_error(self):
+
+    gcs_download_op = components.load_component_from_text("""
```

Comment:
The tests can be made simpler if you just define dummy components at the top and use these in all your pipelines. Dummy Python func based components should suffice for the tests (no need for all this boilerplate).